test/System/String.test.cpp

```diff
@@ -1,8 +1,401 @@
+            CHECK(string.toUtf16() == std::basic_string<std::uint16_t>{0xDBEA, 0xDFCD, 'r', 's'});
+            CHECK(string.toUtf32() == std::basic_string<std::uint32_t>{0x10ABCD, 'r', 's'});
+            CHECK(string.getSize() == 3);
+            CHECK(!string.isEmpty());
+            CHECK(string.getData() != nullptr);
+        }
+
+        SUBCASE("UTF-32 string constructor")
+        {
+            const sf::String string(std::basic_string<std::uint32_t>{'t', 'u', 'v', 0x104321});
+            CHECK(std::string(string) == "tuv\0"s);
+            CHECK(std::wstring(string) == select(L"tuv"s, L"tuv\U00104321"s));
+            CHECK(string.toAnsiString() == "tuv\0"s);
+            CHECK(string.toWideString() == select(L"tuv"s, L"tuv\U00104321"s));
+            CHECK(string.toUtf8() == std::basic_string<std::uint8_t>{'t', 'u', 'v', 0xF4, 0x84, 0x8C, 0xA1});
+            CHECK(string.toUtf16() == std::basic_string<std::uint16_t>{'t', 'u', 'v', 0xDBD0, 0xDF21});
+            CHECK(string.toUtf32() == std::basic_string<std::uint32_t>{'t', 'u', 'v', 0x104321});
+            CHECK(string.getSize() == 4);
+            CHECK(!string.isEmpty());
+            CHECK(string.getData() != nullptr);
+        }
+    }
+
+    SUBCASE("fromUtf8()")
+    {
+        constexpr std::array<std::uint8_t, 4> characters{'w', 'x', 'y', 'z'};
+        const sf::String                      string = sf::String::fromUtf8(characters.begin(), characters.end());
+        CHECK(std::string(string) == "wxyz"s);
+        CHECK(std::wstring(string) == L"wxyz"s);
+        CHECK(string.toAnsiString() == "wxyz"s);
+        CHECK(string.toWideString() == L"wxyz"s);
+        CHECK(string.toUtf8() == std::basic_string<std::uint8_t>{'w', 'x', 'y', 'z'});
+        CHECK(string.toUtf16() == std::basic_string<std::uint16_t>{'w', 'x', 'y', 'z'});
+        CHECK(string.toUtf32() == std::basic_string<std::uint32_t>{'w', 'x', 'y', 'z'});
+        CHECK(string.getSize() == 4);
+        CHECK(!string.isEmpty());
+        CHECK(string.getData() != nullptr);
+    }
+
+    SUBCASE("fromUtf16()")
+    {
+        constexpr std::array<std::uint16_t, 4> characters{'w', 'x', 'y', 'z'};
+        const sf::String                       string = sf::String::fromUtf16(characters.begin(), characters.end());
+        CHECK(std::string(string) == "wxyz"s);
+        CHECK(std::wstring(string) == L"wxyz"s);
+        CHECK(string.toAnsiString() == "wxyz"s);
+        CHECK(string.toWideString() == L"wxyz"s);
+        CHECK(string.toUtf8() == std::basic_string<std::uint8_t>{'w', 'x', 'y', 'z'});
+        CHECK(string.toUtf16() == std::basic_string<std::uint16_t>{'w', 'x', 'y', 'z'});
+        CHECK(string.toUtf32() == std::basic_string<std::uint32_t>{'w', 'x', 'y', 'z'});
+        CHECK(string.getSize() == 4);
+        CHECK(!string.isEmpty());
+        CHECK(string.getData() != nullptr);
+    }
+
+    SUBCASE("fromUtf32()")
+    {
+        constexpr std::array<std::uint32_t, 4> characters{'w', 'x', 'y', 'z'};
```

Comment:
Would it make sense to test non-ascii characters here?

Reply:
Yeah I think I just overlooked that when I was added on-ASCII characters to the various constructors.